ESMF/GOCART2G_GridComp/DU2G_GridComp/DU2G_GridCompMod.F90

```diff
@@ -137,6 +150,8 @@ subroutine SetServices (GC, RC)
     call ESMF_ConfigGetAttribute (cfg, self%Ch_DU_res,  label='Ch_DU:', __RC__)
     call ESMF_ConfigGetAttribute (cfg, self%rlow,       label='radius_lower:', __RC__)
     call ESMF_ConfigGetAttribute (cfg, self%rup,        label='radius_upper:', __RC__)
+    call ESMF_ConfigGetAttribute (cfg, emission_scheme, label='emission_scheme:', default='ginoux', __RC__)
```

Comment:
Alas,   I think the string must be `ESMF_MAXLEN`  for this call to be safe.

I'd otherwise be pushing to make `emission_scheme` a deferred length string.

Reply:
Looking at the rest of the code, I see significant benefit to letting the main variable be deferred:
```f90
character(len=:), allocatable :: emission_scheme
```
You would then have a tmp var of size ESMF_MAXLEN for the GetAttribute() call.   And then you can skip all the trims after the assignment
```
emission_scheme = trim(tmp_buffer)
```

Reply:
Implemented in commit ac12571cb7feb210361bc99519199acfee575d79